libs/blocks/merch/merch.js

```diff
@@ -1,243 +1,138 @@
+    displayTax,
+    template,
+  };
+}
+
+export async function buildCta(el, params) {
+  const context = await getCheckoutContext(el, params);
+  if (!context) return null;
+  const service = await initService();
+  const cta = service.createCheckoutLink(
+    context,
+    el.textContent?.replace(/^CTA +/, ''),
+  );
+  cta.classList.add('con-button');
+  if (el.closest('.marquee')) {
+    cta.classList.add('button-l');
   }
-  const { searchParams } = new URL(el.href);
-  const osi = searchParams.get('osi');
-  const type = searchParams.get('type');
-  if (!(osi && type)) {
-    el.textContent = '';
-    el.setAttribute('aria-details', 'Invalid merch block');
-    return undefined;
+  if (el.firstElementChild?.tagName === 'STRONG' || el.parentElement?.tagName === 'STRONG') {
+    cta.classList.add('blue');
   }
+  return cta;
+}
 
-  const promotionCode = (searchParams.get('promo')
-      ?? el.closest('[data-promotion-code]')?.dataset.promotionCode)
-    || undefined;
-
-  const perpetual = searchParams.get('perp') === 'true' || undefined;
+async function buildPrice(el, params) {
+  const context = await getPriceContext(el, params);
+  if (!context) return null;
+  const service = await initService();
+  const price = service.createInlinePrice(context);
+  return price;
+}
 
-  if (isCTA(type)) {
-    const options = omitNullValues({
-      promotionCode,
-      perpetual,
-      wcsOsi: osi,
-      ...getCheckoutContext(searchParams, getConfig()),
-    });
-    const cta = buildCheckoutButton(el, options);
-    const parentElem = el.parentNode;
-    el.replaceWith(cta);
-    decorateLinkAnalytics(parentElem);
-    return cta;
+export async function polyfill() {
+  if (polyfill.done) return;
+  polyfill.done = true;
+  let isSupported = false;
+  document.createElement('div', {
+    // eslint-disable-next-line getter-return
+    get is() {
+      isSupported = true;
+    },
+  });
+  /* c8 ignore start */
+  if (!isSupported) {
+    const { codeRoot, miloLibs } = getConfig();
+    const base = miloLibs || codeRoot;
+    await loadScript(`${base}/deps/custom-elements.js`);
   }
+  /* c8 ignore stop */
+}
 
-  const displayRecurrence = searchParams.get('term');
-  const displayPerUnit = searchParams.get('seat');
-  const displayTax = searchParams.get('tax');
-  const displayOldPrice = promotionCode ? searchParams.get('old') : undefined;
-  const price = buildPrice({
-    wcsOsi: osi,
-    template: type === 'price' ? undefined : type,
-    displayRecurrence,
-    displayPerUnit,
-    displayTax,
-    displayOldPrice,
-    promotionCode,
-    perpetual,
-  });
-  el.replaceWith(price);
-  return price;
+export default async function init(el) {
+  if (!el?.classList?.contains('merch')) return undefined;
+  await polyfill();
+  const { searchParams: params } = new URL(el.href);
+  const service = await initService();
+  const log = service.log.module('merch');
+  const build = params.get('type') === 'checkoutUrl' ? buildCta : buildPrice;
```

Comment:
variable name `build` is confusing.  Can we give an explicit name such as `merchElement` or `merchEl` or just `el`.